tests/ModelTest.php

```diff
@@ -1151,4 +1151,21 @@ public function testUpdateOrCreate(array $criteria)
         $this->assertEquals($createdAt, $checkUser->created_at->getTimestamp());
         $this->assertEquals($updatedAt, $checkUser->updated_at->getTimestamp());
     }
+
+    public function testCreateWithNullId()
+    {
+        $user = User::create(['_id' => null, 'email' => 'foo@bar']);
+        $this->assertNotNull(ObjectId::class, $user->id);
+        $this->assertSame(1, User::count());
+    }
+
+    public function testUpdateOrCreateWithNullId()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('You must provide attributes to check for duplicates');
+        User::updateOrCreate(
+            ['_id' => null],
```

Comment:
Do users typically pass an explicit `null` value to this method, or would this argument come from another source such as an empty model object?

Reply:
From the [original issue](https://github.com/mongodb/laravel-mongodb/issues/2945), `null` is a fallback value.

```php
$id = $request->get('id') ?? null;
Model::updateOrCreate(['_id' => $id], $data);
```